### DISCUSSION TITLE
Apply one profile multiple times

### DISCUSSION THREAD

**Discussion by MathemanFlo** [libvips/libvips#2041] [Q&A]
Hello,
currently I'm converting cmyk images to rgb using `vips_icc_transform`. It works like a charm and everything is fine besides the need to read the two profiles over and over again for each conversion. So my question is: Is there a way to load the icc profiles once and do the conversions later on without reloading the profile? The only function I could find was `vips_profile_load `which does exactly what I'd like to archieve but could not find a function to do the actual conversion with two profiles as `VipsBlob`.
Thank you very much for your help
Florian

**Comment by jcupitt** [comment]
Hello @MathemanFlo,

libvips should be caching the loaded profiles, so hopefully the load part only happens once.

It will call `cmsOpenProfileFromMem()` many times. I don't know how expensive this operation is. We'd need to do some profiling to see if it appears in the function list.
